README: add that beanstalkd requires C99 compatible compiler

Keith, one question. Did you build beanstalkd with C89 in mind? I am asking because I have merged some changes that do not conform to C89 and not sure if I have made a mistake there. I just realized that you were following C89 with variable declaration. I could not find anything mentioned in this repo or in docs about C89. But I see that we use types like uint32_t, then most probably compiler should be "modern", at least C99 and since we are not doing OS stuff here, we might be safe to assume C99 support.

I see that great C programs (git, SQLite) are C89 compilers compatible. For Linux Kernel it seems very important, but I am not sure if there is a need for program like beanstalkd.

What is your opinion on this matter?

Alright, another data point in this saga:

Trying to switch on Wextra lead to some weird errors from gcc 4.8.* when optimization is turned on. Right now I am thinking that C99 is a bad thing. Even when there are designated initializer all over the place, some old compilers complain about some fields. This is the example:
```
static int
mustdiallocal(int port)
{
    struct sockaddr_in addr = {
        .sin_family = AF_INET,
        .sin_port = htons(port),
    };
```
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4:
```
testserv.c: In function ‘mustdiallocal’:
testserv.c:66:5: error: missing initializer for field ‘sin_addr’ of ‘struct sockaddr_in’ [-Werror=missing-field-initializers]
     };
     ^
In file included from /usr/include/netdb.h:27:0,
                 from testserv.c:11:
/usr/include/netinet/in.h:241:20: note: ‘sin_addr’ declared here
     struct in_addr sin_addr;  /* Internet address.  */
                    ^
```
I am starting to think about compiling the code with gnu89. Am I overthinking this?

--- a/README.md
+++ b/README.md
@@ -30,7 +30,10 @@ also try,
     $ make install
     $ make install PREFIX=/usr
 
-Requires Linux (2.6.17 or later), Mac OS X, or FreeBSD.
+Requires Linux (2.6.17 or later), Mac OS X, FreeBSD, or Illumos.
+
+Currently beanstalkd is tested with GCC and clang, but it should work
+with any compiler that supports C99.
 
 Uses ronn to generate the manual.
 See http://github.com/rtomayko/ronn.

